### DISCUSSION TITLE
Artifactory Works with Cargo, Fails with Bazel

### DISCUSSION THREAD

**Discussion by ericmcbride** [bazelbuild/rules_rust#2484] [General]
I currently have cargo working with pulling crates from artifactory for a private crates registry.  Im having issues running
```
CARGO_REGISTRIES_MY_REGISTRY_TOKEN="1234" CARGO_BAZEL_ISOLATED=0 CARGO_HOME="~/.cargo" CARGO_BAZEL_REPIN=1 bazel sync --only=crates_index
```
Reason im setting `CARGO_HOME` and `CARGO_BAZEL_ISOLATED` is because i looked thru the source code and saw it was using crates_index which relies on `CARGO_HOME`

The error im receiving is:
```
Error: Failed to get crate index config
Caused by:
    No such file or directory (os error 2)
```
Which is telling me it cant find my `CARGO_HOME/registry/index/myregistry-1234/config.json`

Example cargo/config.toml
```
// cargo/config.toml 
[registries.myregistry]
index="sparse+https://artifactory-my-domain/artifactory/my-local-sparse-index/index/"

[source.myregistry]
registry = "sparse+https://artifactory-my-domain/artifactory/my-local-sparse-index/index/"

[registry]
default = "myregistry"
global-credential-providers = ["cargo:token"]

[net]
git-fetch-with-cli = true
```
Any help would be greatly appreciated.

Message rust

**Comment by ramzesucr** [comment]
Did you found workaround?

**Reply by ericmcbride** [reply]
There were a couple issues.  Upstream the  index config was in the wrong spot.  2nd off I Had to move my local `.cargo/credentials.toml` to my global `~/.cargo/credentials.toml` for my credentials to work (also check ur .netrc is being populated with creds).  I moved off WORKSPACE to Module.bazel as well so I didn't have to do the sync anymore.

**Reply by MarcelGep** [reply]
Can you explain in detail how you fix this problem? I use from_cargo and apply the cargo config.toml file with the parameter cargo_config but there is no parameter for credentials.toml.

**Reply by UebelAndre** [reply]
You'll likely also need `isolated = False` or `CARGO_BAZEL_ISOLATED=0` (env var).

**Reply by MarcelGep** [reply]
i set `isolated = False` on from_cargo

**Reply by MarcelGep** [reply]
with a .netrc in my home directory it works fine but why it does't work with the cargo credentials?
